### ISSUE TITLE
[Bug]: AnimateDiff goes away from the seed number

### ISSUE THREAD

**Issue by 3dcinetv** [continue-revolution/sd-webui-animatediff#270]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### Have you read FAQ on README?

- [X] I have updated WebUI and this extension to the latest version

### What happened?

Same seed is not used when animatediff is active. Animate diff uses (a non prompted text, non prompted seed) and does whatever other generation.

### Steps to reproduce the problem

When I write my prompt for a human to wave the right hand in the format:
(my prompt), masterpiece, high quality, Lora:1
0: open mouth
24: wave right hand
Size: 416x720 (upscaled x2: 832x1440)
Seed: 1746215953
48 frames, 8fps

It generates a totally different (gif), png sequence, than Stable diffusion generated when AnimateDiff was off.
If I have a really close shot of the person doing the wave, AnimateDiff presents the entire body at the moment of generating the .gif.

Please let me know if there's something I need to modify in the PROMPT, on the SIZE or in the FPS in order to get the same thing I generated without AnimateDiff active.

I got the same image SD generated and good result from seed number in another experiment (size: 416x768), with the same format than the above prompt. 
Please help.

### What should have happened?

Same seed should generate same image when using AnimDiff

### Commit where the problem happens

version: [v1.6.0](https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/5ef669de080814067961f28357256e8fe27544f4)  •  python: 3.10.6  •  torch: 2.0.1+cu118  •  xformers: 0.0.20  •  gradio: 3.41.2  •  checkpoint: [4d2cbbb94f]

### Command Line Arguments

```Shell
no
```

### Console logs

```Shell
no
```

**Comment by 3dcinetv** [comment]
Here's the LOG:
 Traceback (most recent call last):
      File "E:\AI_anime_SDv4\stable-diffusion-webui\modules\call_queue.py", line 57, in f
        res = list(func(*args, **kwargs))
      File "E:\AI_anime_SDv4\stable-diffusion-webui\modules\call_queue.py", line 36, in f
        res = func(*args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\modules\txt2img.py", line 55, in txt2img
        processed = processing.process_images(p)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\modules\processing.py", line 732, in process_images
        res = process_images_inner(p)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\extensions\sd-webui-animatediff\scripts\animatediff_cn.py", line 119, in hacked_processing_process_images_hijack
        return getattr(processing, '__controlnet_original_process_images_inner')(p, *args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\modules\processing.py", line 867, in process_images_inner
        samples_ddim = p.sample(conditioning=p.c, unconditional_conditioning=p.uc, seeds=p.seeds, subseeds=p.subseeds, subseed_strength=p.subseed_strength, prompts=p.prompts)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\modules\processing.py", line 1156, in sample
        return self.sample_hr_pass(samples, decoded_samples, seeds, subseeds, subseed_strength, prompts)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\modules\processing.py", line 1242, in sample_hr_pass
        samples = self.sampler.sample_img2img(self, samples, noise, self.hr_c, self.hr_uc, steps=self.hr_second_pass_steps or self.steps, image_conditioning=image_conditioning)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 188, in sample_img2img
        samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args=self.sampler_extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
      File "E:\AI_anime_SDv4\stable-diffusion-webui\modules\sd_samplers_common.py", line 261, in launch_sampling
        return func()
      File "E:\AI_anime_SDv4\stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 188, in <lambda>
        samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args=self.sampler_extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
        return func(*args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\repositories\k-diffusion\k_diffusion\sampling.py", line 145, in sample_euler_ancestral
        denoised = model(x, sigmas[i] * s_in, **extra_args)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
        return forward_call(*args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\extensions\sd-webui-animatediff\scripts\animatediff_infv2v.py", line 277, in mm_cfg_forward
        x_out = mm_sd_forward(self, x_in, sigma_in, cond_in, image_cond_in, make_condition_dict) # hook
      File "E:\AI_anime_SDv4\stable-diffusion-webui\extensions\sd-webui-animatediff\scripts\animatediff_infv2v.py", line 187, in mm_sd_forward
        out = self.inner_model(
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
        return forward_call(*args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\repositories\k-diffusion\k_diffusion\external.py", line 112, in forward
        eps = self.get_eps(input * c_in, self.sigma_to_t(sigma), **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\repositories\k-diffusion\k_diffusion\external.py", line 138, in get_eps
        return self.inner_model.apply_model(*args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\modules\sd_hijack_utils.py", line 17, in <lambda>
        setattr(resolved_obj, func_path[-1], lambda *args, **kwargs: self(*args, **kwargs))
      File "E:\AI_anime_SDv4\stable-diffusion-webui\modules\sd_hijack_utils.py", line 28, in __call__
        return self.__orig_func(*args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\models\diffusion\ddpm.py", line 858, in apply_model
        x_recon = self.model(x_noisy, t, **cond)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
        return forward_call(*args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\models\diffusion\ddpm.py", line 1335, in forward
        out = self.diffusion_model(x, t, context=cc)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
        return forward_call(*args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\modules\sd_unet.py", line 91, in UNetModel_forward
        return ldm.modules.diffusionmodules.openaimodel.copy_of_UNetModel_forward_for_webui(self, x, timesteps, context, *args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\modules\diffusionmodules\openaimodel.py", line 797, in forward
        h = module(h, emb, context)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
        return forward_call(*args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\modules\diffusionmodules\openaimodel.py", line 86, in forward
        x = layer(x)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
        return forward_call(*args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\extensions\sd-webui-animatediff\motion_module.py", line 107, in forward
        return self.temporal_transformer(input_tensor, encoder_hidden_states, attention_mask)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
        return forward_call(*args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\extensions\sd-webui-animatediff\motion_module.py", line 173, in forward
        hidden_states = block(hidden_states, encoder_hidden_states=encoder_hidden_states, video_length=video_length)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
        return forward_call(*args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\extensions\sd-webui-animatediff\motion_module.py", line 237, in forward
        hidden_states = attention_block(
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
        return forward_call(*args, **kwargs)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\extensions\sd-webui-animatediff\motion_module.py", line 596, in forward
        hidden_states = self._memory_efficient_attention(query, key, value, attention_mask, optimizer_name)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\extensions\sd-webui-animatediff\motion_module.py", line 494, in _memory_efficient_attention
        hidden_states = xformers.ops.memory_efficient_attention(
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\xformers\ops\fmha\__init__.py", line 192, in memory_efficient_attention
        return _memory_efficient_attention(
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\xformers\ops\fmha\__init__.py", line 290, in _memory_efficient_attention
        return _memory_efficient_attention_forward(
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\xformers\ops\fmha\__init__.py", line 310, in _memory_efficient_attention_forward
        out, *_ = op.apply(inp, needs_gradient=False)
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\xformers\ops\fmha\cutlass.py", line 175, in apply
        out, lse, rng_seed, rng_offset = cls.OPERATOR(
      File "E:\AI_anime_SDv4\stable-diffusion-webui\venv\lib\site-packages\torch\_ops.py", line 502, in __call__
        return self._op(*args, **kwargs or {})
    RuntimeError: CUDA error: invalid configuration argument
    CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
    For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
    Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

**Comment by zappityzap** [comment]
The options mentioned here may resolve the CUDA error: https://github.com/continue-revolution/sd-webui-animatediff/issues/268#issuecomment-1789711775

**Comment by continue-revolution** [comment]
This cuda error is simply because of xformers. It has nothing to do with the cuda error in another thread. Please search for issues or read the update message v1.5.2 to see what you can do.

It is normal to get totally different image if you add AnimateDiff, just like what will happen if you enable CN with the same seed. AnimateDiff is a plug-in model which can affect generation for each frame.

**Comment by 3dcinetv** [comment]
"AnimateDiff is a plug-in model which can affect generation for each frame." --->> how then, do I generate a previously prompted image if seed number is not obeyed by AnimateDiff?

**Comment by continue-revolution** [comment]
If you use the same seed for 2 gif generation, you will get the same gif. If you want to use seed to recover a single image, unfortunately there is no way to do that.

**Comment by 3dcinetv** [comment]
My current workflow (where the issue appears) is:
1. I define the size, steps and CFG scale. I generate a random image with my custom prompt
2. I get a seed number. This ensures my image would be the same, right?
3. Wrong: when I activate AnimateDiff with the "mm_sd_v15_v2" model, I get a totally different image (same character, just not in the position I got it on the first time when I generated it from the original seed).
4. My settings in AnimDiff: Number of Frames: 72, 8fps, InterpX 10. Save format (.PNG only).
My current size is 720x516

Hence, is there a way to get the same character position in the generated images as the original seed number? or will AnimateDiff always change to (any random) seed?
